pkg/roachpb/api.proto

```diff
@@ -384,6 +384,10 @@ message DeleteRangeRequest {
   // considers empty spans equivalent to being covered by an MVCC range
   // tombstone, so it will omit the write across an entirely empty span too.
   bool idempotent_tombstone = 7;
+  // If set and DeleteRangeRequest is deleting the whole range with a range
+  // tombstone, gc_range_hint will update replica GC hint to notify mvcc gc
+  // queue that this range would benefit from optimized GC process.
+  GCRangeHint gc_range_hint = 8;
```

Comment:
I wouldn't send the entire GC hint explicitly here, since we may not want callers to get access to everything in that struct. I'd add a boolean field, something like `AggressiveGC` or `IsSchemaDrop` or something, and then set the GC hint to the current timestamp based on that.